api_v3/lib/types/KalturaUserEntry.php

```diff
@@ -159,4 +159,68 @@ protected function doFromObject($srcObj, KalturaDetachedResponseProfile $respons
 		parent::doFromObject($srcObj, $responseProfile);
 	}
 
+	public function validateForInsert($propertiesToSkip = array())
+	{
+		parent::validateForInsert($propertiesToSkip);
+		$this->validatePropertyNotNull("entryId");
+		$this->validateEntryId();
+		$this->validateUserID();
+	}
+	
+	public function validateForUpdate($sourceObject, $propertiesToSkip = array())
+	{
+		if($this->entryId !== null)
+			$this->validateEntryId($sourceObject->getId());
+
+		$this->validateUserID($sourceObject->getId());
+		return parent::validateForUpdate($sourceObject, $propertiesToSkip);
+	}
+	
+	/*
+	 * @param string $userEntryID
+	 * @throw KalturaAPIException
+	 */
+	private function validateEntryId($userEntryID = null)
+	{
+		if($userEntryID === null)
+		{
+			$dbEntry = entryPeer::retrieveByPK($this->entryId);
+			if (!$dbEntry)
+				throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $this->entryId);
+		}
+		else
+		{
+			// update
+			$dbUserEntry = UserEntryPeer::retrieveByPK($userEntryID);
+			if(!$dbUserEntry)
+				throw new KalturaAPIException(KalturaErrors::USER_ENTRY_NOT_FOUND, $userEntryID);
+				
+			if($this->entryId !== null && $this->entryId != $dbUserEntry->getEntryId())
+				throw new KalturaAPIException(KalturaErrors::USER_ENTRY_DOES_NOT_MATCH_ENTRY_ID, $userEntryID);
+		}
+	}
+	
+	/*
+	 * @param string $userEntryID
+	 * @throw KalturaAPIException
+	 */
+	private function validateUserId($userEntryID = null)
+	{
+		if($userEntryID === null)
+		{
+			$userId = $this->userId ? $this->userId : kCurrentContext::getCurrentKsKuserId();
+			if(!$userId)
+				throw new KalturaAPIException(KalturaErrors::USER_ID_NOT_PROVIDED);
+		}
+		else
```

Comment:
@yossipapi don't think you need this else block. userId property on the KalturaUserEntry object is "@insertonly" and cannot be updated.